src/executor/current_thread/mod.rs

```diff
@@ -132,6 +133,12 @@ pub struct CurrentThread<P: Park = ParkThread> {
 
     /// Thread park handle
     park: P,
+
+    /// Handle for spawning new futures from other threads
+    spawn_handle: Handle,
+
+    /// Receiver for futures spawned from other threads
+    spawn_receiver: executor::Spawn<mpsc::UnboundedReceiver<Box<Future<Item = (), Error = ()> + Send + 'static>>>
```

Comment:
This can probably be a plain old queue. There is no need to pull in all the future enabled stuff since this is for an executor. The easiest thing here is probably to use `std::sync::mpsc` and `try_recv`.

Reply:
Thanks, it indeed is much simpler with a normal mpsc queue. Changed it accordingly